utils/clang-tidy-diff.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python2
+#
+#===- clang-tidy-diff.py - ClangTidy Diff Checker ------------*- python -*--===#
+#
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+#
+#===------------------------------------------------------------------------===#
+
+# Changes made to the initial version from
```

Comment:
Why do we need to modify the original script? I think we can use it as is.

Reply:
clang-tidy does not “run” on the header files (as they aren’t part of the .compile_commands.json file), but it will emit warning for headers 1) if the header file matches the `-header-filter` regex option and 2) a C/C++ source file including that header file is being compiled.

The problem with the original script, is that it runs only on the modified files, so if only a header file is modified, the original script will never tell you there could be an issue with your modifications. Hence why I modified it to have it run on all files, but only report warnings/errors on the files that were modified.